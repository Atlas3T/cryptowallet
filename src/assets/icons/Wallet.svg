<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="0 0 51 51" style="enable-background:new 0 0 51 51;" xml:space="preserve">
<style type="text/css">
	.st0{fill:none;stroke:#000000;stroke-width:5;stroke-miterlimit:10;}
	.st1{display:none;}
	.st2{display:inline;fill:none;stroke:#000000;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
	.st3{display:inline;}
</style>
<g id="Wallet">
	<path class="st0" d="M46.1,31.5v5.3c0,2.4-1.9,4.3-4.3,4.3H9.5c-2.4,0-4.3-1.9-4.3-4.3V15.5c0-2.4,1.9-4.3,4.3-4.3h32.3
		c2.4,0,4.3,1.9,4.3,4.3v5.2"/>
	<circle cx="46.1" cy="26.1" r="3.1"/>
</g>
<g id="Payments" class="st1">
	<line class="st2" x1="8" y1="19.8" x2="41.8" y2="19.8"/>
	<polyline class="st2" points="35.8,13.3 42.3,19.8 35.8,26.3 	"/>
	<line class="st2" x1="42.3" y1="32.8" x2="8.5" y2="32.8"/>
	<polyline class="st2" points="14.5,39.3 8,32.8 14.5,26.3 	"/>
</g>
<g id="Exchange" class="st1">
	<path class="st2" d="M15.3,21.7c0-5.6,4.6-10.2,10.2-10.2s10.2,4.6,10.2,10.2"/>
	<polyline class="st2" points="42.2,16.3 35.7,22.8 29.2,16.3 	"/>
	<path class="st2" d="M35.7,30.2c0,5.6-4.6,10.2-10.2,10.2s-10.2-4.6-10.2-10.2"/>
	<polyline class="st2" points="8.8,35.7 15.3,29.2 21.9,35.7 	"/>
</g>
<g id="More" class="st1">
	<circle class="st3" cx="26" cy="15.8" r="7.8"/>
	<circle class="st3" cx="12.3" cy="28.6" r="5"/>
	<circle class="st3" cx="35.3" cy="33.6" r="3.5"/>
</g>
</svg>
