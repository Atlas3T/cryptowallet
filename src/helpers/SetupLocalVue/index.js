import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
