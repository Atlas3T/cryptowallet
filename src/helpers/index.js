import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
