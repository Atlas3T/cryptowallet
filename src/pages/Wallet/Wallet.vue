<template>
  <div class="layout">
    <div class="list">
      <WalletsList
        :coins="coins"
        :currency="currency"
      />
    </div>
    <div class="button">
      <AddWallet/>
    </div>
  </div>
</template>

<script>
import WalletsList from '@/components/Wallet/Wallet/WalletsList.vue';
import AddWallet from '@/components/Wallet/Wallet/AddWallet.vue';

export default {
  name: 'Balance',
  components: {
    WalletsList,
    AddWallet,
  },
  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      currency: 'GPB',
      coins: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
