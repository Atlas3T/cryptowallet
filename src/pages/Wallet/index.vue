<template>
  <div class="layout">
    <div class="list">
      <wallet-list
        :wallets="wallets"
        :currency="selectedCurrency"
      />
    </div>
    <div class="button">
      <add-wallet/>
    </div>
  </div>
</template>

<script>
import WalletList from '@/components/Wallet/WalletsList';
import AddWallet from '@/components/Wallet/AddWallet';

export default {
  name: 'Balance',

  components: {
    WalletList,
    AddWallet,
  },

  /**
   * @TODO James Konrad
   * data below is mocked
   *
   * currency String is a prefered user currency, should be in the Vuex store
   * you should probably select your prefered currency during the initial setup
   * alongside the language
   *
   * coins Object also shoudn't be here
   */
  data() {
    return {
      wallets: {
        BTC: {
          key: 'BTC',
          name: 'Bitcoin',
          balance: 25.532344,
          uid: 1,
        },
        ATL: {
          key: 'ATL',
          name: 'Atlas',
          balance: 21.532344,
          uid: 2,
        },
        ETH: {
          key: 'ETH',
          name: 'Ethereum',
          balance: 55.532344,
          uid: 3,
        },
      },
    };
  },

  computed: {
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
  },
};

</script>

<style scoped>
.layout {
  display: flex;
  flex-direction: column;
}

.list {
  flex: 1;
  position: relative;
}
</style>
