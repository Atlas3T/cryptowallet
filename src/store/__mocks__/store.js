import Vue from 'vue';
import Vuex from 'vuex';
import VuexORM from '@vuex-orm/core';
import VuexORMLoki from '@atlascity/vuex-orm-plugin-lokijs';

import wallet from '@/store/wallet';
import contacts from '@/store/contacts/__mocks__/contacts.js';
import search from '@/store/search/__mocks__/search.js';
import settings from '@/store/settings/__mocks__/settings.js';
import setup from '@/store/setup/__mocks__/setup.js';
import qrcode from '@/store/qrcode/__mocks__/qrcode.js';
import modals from '@/store/modals/__mocks__/modals.js';

// entities
import Account from '@/store/wallet/entities/account';
import Address from '@/store/wallet/entities/address';
import Tx from '@/store/wallet/entities/tx';
import Utxo from '@/store/wallet/entities/utxo';
import Wallet from '@/store/wallet/entities/wallet';
import Coin from '@/store/wallet/entities/coin';
import KeyPair from '@/store/wallet/entities/keyPair';
import Prices from '@/store/prices';
import LatestPrice from '@/store/latestPrice';

Vue.use(Vuex);

// Setup ORM database.
const database = new VuexORM.Database();

database.register(Account, {});
database.register(Address, {});
database.register(Tx, {});
database.register(Utxo, {});
database.register(Wallet, {});
database.register(Coin, {});
database.register(KeyPair, {});
database.register(Prices, {});
database.register(LatestPrice, {});

Vue.prototype.encryptedModels = [
  Account,
  Wallet,
];

const options = {
  env: 'browser',
};

VuexORM.use(VuexORMLoki, { database, options });

export const actions = {};
export const getters = {};
export const mutations = {};
export const state = {};

export function createMocks(custom = {
  getters: {}, mutations: {}, actions: {}, state: {},
}) {
  /**
   * mock store modules.
   */
  const modules = {
    wallet,
    search,
    settings,
    contacts,
    setup,
    qrcode,
    modals,
  };

  let mockGettersCombined = {};
  let mockMutationsCombined = {};
  let mockActionsCombined = {};
  let mockStateCombined = {};

  Object.entries(modules).forEach(([name, module]) => {
    mockGettersCombined = Object.assign({}, mockGettersCombined, module.getters);
    mockMutationsCombined = Object.assign({}, mockMutationsCombined, module.mutations);
    mockActionsCombined = Object.assign({}, mockActionsCombined, module.actions);
    mockStateCombined = Object.assign({}, mockStateCombined, module.state);

    if (custom.state && custom.state[name]) {
      module.state = Object.assign(module.state, custom.state[name]);
    }

    modules[name] = {
      namespaced: true,
      getters: Object.assign({}, getters, module.getters),
      mutations: Object.assign({}, mutations, module.mutations),
      actions: Object.assign({}, actions, module.actions),
      state: Object.assign({}, state, module.state),
    };
  });

  mockGettersCombined = Object.assign({}, mockGettersCombined, custom.getters);
  mockMutationsCombined = Object.assign({}, mockMutationsCombined, custom.mutations);
  mockActionsCombined = Object.assign({}, mockActionsCombined, custom.actions);
  mockStateCombined = Object.assign({}, mockStateCombined, custom.state);
  return {
    getters: mockGettersCombined,
    mutations: mockMutationsCombined,
    actions: mockActionsCombined,
    state: mockStateCombined,
    store: new Vuex.Store({
      getters: mockGettersCombined,
      mutations: mockMutationsCombined,
      actions: mockActionsCombined,
      state: mockStateCombined,
      modules,
      plugins: [VuexORM.install(database)],
    }),
  };
}

export const { store } = createMocks();
