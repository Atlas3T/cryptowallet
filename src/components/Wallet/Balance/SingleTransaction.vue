<template>
  <q-timeline-entry
    :title="paymentDirection"
    side="left"
  >
    <q-collapsible
      :label="date"
      :sublabel="amount"
      :class="{ 'positive-amount': data.amount > 0, 'negative-amount': data.amount < 0 }"
    >
      <div>
        <p>
          {{ $t('received') }}:<br>
          <span class="received">{{ data.received }}</span>
        </p>
        <p>
          TX Hash:<br>
          <span class="tx-hash">{{ data.tx_hash }}</span>
        </p>
        <p>
          {{ $t('status') }}:<br>
          <span
            :class="{ 'unconfirmed-tx': data.status === 1, 'confirmed-tx': data.status === 2 }"
            class="status"
          >
            {{ $t(status) }}
          </span>
        </p>
        <p>
          TX {{ $t('fee') }}:<br>
          <span class="tx-fee">{{ data.tx_fee }}</span>
        </p>
      </div>
    </q-collapsible>
  </q-timeline-entry>
</template>

<script>
import { amount, date } from '@/helpers';

export default {
  name: 'SingleTransaction',
  props: {
    data: {
      type: Object,
      required: true,
    },
    currency: {
      type: String,
      default() {
        return '';
      },
    },
  },
  computed: {
    /**
     * Prepends "To: " or "From: " into transaction title
     * @return {String}
     */
    paymentDirection() {
      if (this.data.to) return `${this.$t('to')}: ${this.data.to}`;
      return `${this.$t('from')}: ${this.data.from}`;
    },
    /**
     * Returns formatted date of the transaction
     * @return {String}
     */
    date() {
      return date(this.data.ts, 'DD MMMM', this);
    },
    /**
     * Converts amount value (Number) into a String with optional currency symbol
     * @return {String}
     */
    amount() {
      return amount(this.data.amount, this.currency);
    },
    /**
     * Returns the transaction status in a human friendly format
     * @return {String}
     */
    status() {
      if (this.data.status === 1) return 'unconfirmed';
      if (this.data.status === 2) return 'confirmed';
      return 'pending';
    },
  },
};
</script>

<style>
  .confirmed-tx {
    color: green;
  }

  .unconfirmed-tx {
    color: red;
  }

  .positive-amount .q-item-sublabel {
    color: green;
  }

  .negative-amount .q-item-sublabel {
    color: red;
  }
</style>
