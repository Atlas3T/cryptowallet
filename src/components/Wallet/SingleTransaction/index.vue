<template>
  <q-timeline-entry
    :title="date"
    side="left"
  >
    <q-expansion-item
      :label="amount"
      :class="{ 'positive-amount': !data.sent, 'negative-amount': data.sent }"
    >
      <q-card>
        <q-card-section>
          <div class="single-transaction-content">
            <p>
              TX Hash:
              <span class="tx-hash break">{{ data.hash }}</span>
            </p>
            <p>
              {{ $t('status') }}:
              <span
                :class="{
                  'unconfirmed-tx': data.confirmations < minConfirmations,
                  'confirmed-tx': data.confirmations >= minConfirmations,
                }"
                class="status"
              >
                {{ status }}
              </span>
            </p>
            <p>
              {{ $t('confirmations') }}:
              <span class="confirmations">{{ data.confirmations }}</span>
            </p>
            <p>
              TX {{ $t('fee') }}:
              <span class="tx-fee">{{ feeFormated }}</span>
            </p>
            <p>
              {{ $t('blockHeight') }}:
              <span class="tx-height">{{ data.blockHeight }}</span>
            </p>
          </div>
        </q-card-section>
      </q-card>
    </q-expansion-item>

  </q-timeline-entry>
</template>

<script>
import { mapState } from 'vuex';
import { dateTranslater, AmountFormatter } from '@/helpers';
import Coin from '@/store/wallet/entities/coin';

export default {
  name: 'SingleTransaction',

  props: {
    data: {
      type: Object,
      required: true,
    },
  },

  computed: {
    ...mapState({
      id: (state) => { return state.route.params.id; },
    }),
    wallet() {
      return this.$store.getters['entities/wallet/find'](this.id);
    },
    selectedCurrency() {
      return this.$store.state.settings.selectedCurrency;
    },
    supportedCoins() {
      return Coin.all();
    },
    coinDenomination() {
      return this.supportedCoins.find((coin) => {
        return coin.name === this.wallet.name;
      }).denomination;
    },
    minConfirmations() {
      return this.supportedCoins.find((coin) => {
        return coin.name === this.wallet.name;
      }).minConfirmations;
    },
    coinSymbol() {
      return this.supportedCoins.find((coin) => { return coin.name === this.wallet.name; }).symbol;
    },
    latestPrice() {
      const prices = this.$store.getters['entities/latestPrice/find'](`${this.coinSymbol}_${this.selectedCurrency.code}`);
      if (prices) {
        return prices.data.PRICE;
      }
      return null;
    },

    /**
     * Prepends "To: " or "From: " into transaction title
     * @return {String}
     */
    paymentDirection() {
      if (this.data.sent) {
        let { receiver } = this.data;
        if (Array.isArray(receiver)) { [receiver] = receiver; }
        return `${this.$t('to')}: ${receiver}`;
      }

      let { sender } = this.data;
      if (Array.isArray(sender)) { [sender] = sender; }
      return `${this.$t('from')}: ${sender}`;
    },

    to() {
      let { receiver } = this.data;
      if (Array.isArray(receiver)) { [receiver] = receiver; }
      return receiver;
    },

    /**
     * Returns formatted date of the transaction
     * @return {String}
     */
    date() {
      const msToS = 1000;
      if (this.data.receivedTime) {
        return dateTranslater(new Date(this.data.receivedTime * msToS).valueOf(), 'DD MMMM HH:mm YYYY', this);
      }
      return dateTranslater(new Date(this.data.confirmedTime * msToS).valueOf(), 'DD MMMM HH:mm YYYY', this);
    },

    /**
     * Converts amount value (Number) into a String
     * @return {String}
     */
    amount() {
      const { value } = this.data;
      let inCoin = value;
      if (this.data.sent) { inCoin = -Math.abs(inCoin); }

      const amountInCoin = new AmountFormatter({
        amount: inCoin,
        rate: this.latestPrice,
        format: this.coinDenomination,
        prependPlusOrMinus: true,
        removeTrailingZeros: true,
      });

      if (this.latestPrice) {
        const amountInCurrency = new AmountFormatter({
          amount: inCoin,
          rate: this.latestPrice,
          format: '0,0[.]00',
          coin: this.wallet.name,
          prependPlusOrMinus: false,
          currency: this.selectedCurrency,
          toCurrency: true,
          toCoin: false,
          withCurrencySymbol: true,
        });

        return `${amountInCoin.getFormatted()} ${this.coinSymbol} (${amountInCurrency.getFormatted()})`;
      }
      return `${amountInCoin.getFormatted()} ${this.coinSymbol}`;
    },

    /**
     * Displays the fee
     * @return {String}
     */
    feeFormated() {
      const { fee } = this.data;
      const inCoin = fee;
      if (this.latestPrice) {
        let feeInCoin = new AmountFormatter({
          amount: inCoin,
          rate: this.latestPrice,
          format: this.coinDenomination,
          prependPlusOrMinus: false,
          removeTrailingZeros: true,
        });

        let amountInCurrency = new AmountFormatter({
          amount: feeInCoin.getFormatted(),
          rate: this.latestPrice,
          format: '0,0[.]00',
          coin: this.wallet.name,
          prependPlusOrMinus: false,
          currency: this.selectedCurrency,
          toCurrency: true,
          toCoin: false,
          withCurrencySymbol: true,
        });

        if (this.wallet.sdk === 'ERC20') {
          const parent = this.supportedCoins.find((coin) => {
            return coin.name === this.wallet.parentName;
          });
          const price = this.$store.getters['entities/latestPrice/find'](`${parent.symbol}_${this.selectedCurrency.code}`).data.PRICE;

          feeInCoin = new AmountFormatter({
            amount: inCoin,
            rate: price,
            format: parent.denomination,
            prependPlusOrMinus: false,
            removeTrailingZeros: true,
          });

          amountInCurrency = new AmountFormatter({
            amount: feeInCoin.getFormatted(),
            rate: price,
            format: '0,0[.]00',
            coin: this.wallet.parentName,
            prependPlusOrMinus: false,
            currency: this.selectedCurrency,
            toCurrency: true,
            toCoin: false,
            withCurrencySymbol: true,
          });
          return `${fee} ${parent.symbol} (${amountInCurrency.getFormatted()})`;
        }

        return `${fee} ${this.coinSymbol} (${amountInCurrency.getFormatted()})`;
      }

      return `${fee} ${this.coinSymbol}`;
    },

    /**
     * Returns the transaction status in a human friendly format
     * @return {String}
     */
    status() {
      if (this.data.confirmations > this.minConfirmations) { return this.$t('confirmed'); }
      if (this.data.confirmations > 0) { return this.$t('unconfirmed'); }
      return this.$t('pending');
    },
  },
};
</script>

<style>
.positive-amount .q-item__label,
.status.confirmed-tx {
  color: green;
}

.negative-amount .q-item__label,
.status.unconfirmed-tx {
  color: red;
}

.q-timeline-entry {
  padding-left: 1.5rem;
}

.q-item {
  padding: 0;
}

.q-timeline-content {
  padding-bottom: 0.5rem;
}

.q-timeline-title {
  margin-bottom: 0rem;
}

.q-timeline-dot:after {
  opacity: 0.1;
  left: 7px;
}

.q-timeline-dot:before, .q-timeline-dot:after {
    content: '';
    background: #4e677d;
    display: block;
    position: absolute;
}

.q-timeline-dot:before {
    border: 3px solid transparent;
    border-radius: 100%;
    height: 12px;
    width: 12px;
    top: 5px;
    left: 2px;
    transition: background 0.3s ease-in-out, border 0.3s ease-in-out;
}

.q-timeline {
  padding: 0.5rem;
}

.single-transaction-content {
  font-size: 0.8rem;
}

.q-timeline--dense--right
.q-timeline__entry {
  padding-left: 1.5rem;
}

.q-timeline__title {
  font-size: 1rem;
  padding-top: 0.1rem;
  margin-bottom: 0;
}

.q-expansion-item .q-item__label {
  font-size: 0.8rem;
}

.q-expansion-item .q-item {
  min-height: 1rem;
}

.q-expansion-item .q-card__section {
  padding: 0;
}

.q-expansion-item .q-card__section p {
  margin: 0;
  padding: 0.5rem;
  border-top: 1px solid whitesmoke;
}

.q-expansion-item__container {
  border-top: 1px solid white;
}

.break {
  word-break: break-all;
}
</style>
