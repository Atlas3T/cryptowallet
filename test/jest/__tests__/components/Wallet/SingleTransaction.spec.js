import { shallowMount } from '@vue/test-utils';
import SingleTransaction from '@/components/Wallet/SingleTransaction';
import { localVue, i18n, createRouter } from '@/helpers/SetupLocalVue';
import { createMocks as createStoreMocks } from '@/store/__mocks__/store.js';
import Coin from '@/store/wallet/entities/coin';
import Wallet from '@/store/wallet/entities/wallet';
import LatestPrice from '@/store/latestPrice';
import SupportedCoins from '@/store/settings/state/supportedCoins.js';


describe('SingleTransaction.vue', () => {
  let storeMocks;
  let wrapper;
  let router;

  const propsData = {
    data: {
      amount: 4.97,
      receiver: ['1CK6KHY6MHgYvmRQ4PAafKYDrg1ejbH1cE'],
      hash: 'd02995c1faf572c5d9ae966915aa7958c88ed2e47f8f208b68d82d863bf61010',
      sent: true,
      confirmations: 12,
      blockHeight: 245,
      fee: 0.1,
      receivedTime: '1547746030',
      confirmedTime: '1547746030',
    },
  };


  function wrapperInit(options) {
    SupportedCoins.forEach((coin) => {
      Coin.insert({ data: coin });
    });
    Wallet.insert({
      data: [{
        id: 4,
        account_id: 1,
        name: 'Bitcoin',
        displayName: 'Bitcoin',
        sdk: 'Bitcoin',
      },
      {
        id: 5,
        account_id: 1,
        name: 'Ethereum',
        displayName: 'Ethereum',
        sdk: 'Ethereum',
      },
      {
        id: 6,
        account_id: 1,
        name: 'Catalyst',
        displayName: 'Catalyst',
        parentName: 'Ethereum',
        sdk: 'ERC20',
      },
      {
        id: 7,
        account_id: 1,
        name: 'Litecoin',
        displayName: 'Litecoin',
        sdk: 'Bitcoin',
      }],
    });

    LatestPrice.insert({
      data: [{
        coin: 'BTC',
        currency: 'GBP',
        data: {
          CHANGEPCT24HOUR: 4.208515135911541,
          PRICE: 3845.19,
          VOLUME24HOURTO: 10458846.999825574,
        },
      },
      {
        coin: 'ETH',
        currency: 'GBP',
        data: {
          CHANGEPCT24HOUR: 5.947834565694315,
          PRICE: 130.39,
          VOLUME24HOURTO: 1897390.3205589484,
        },
      }],
    });
    return shallowMount(SingleTransaction, options);
  }

  function storeInit(custom) {
    storeMocks = createStoreMocks(custom);
    router = createRouter(storeMocks.store);
    router.push({ path: '/wallet/single/4' });
    wrapper = wrapperInit({
      i18n, router, localVue, store: storeMocks.store, propsData,
    });
  }

  beforeEach(() => { return storeInit(); });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays a received Bitcoin transaction', () => {
    wrapper.setProps({
      data: {
        amount: 4.97,
        sender: ['1CK6KHY6MHgYvmRQ4PAafKYDrg1ejbH1cE'],
        hash: 'd02995c1faf572c5d9ae966915aa7958c88ed2e47f8f208b68d82d863bf61010',
        sent: false,
        confirmations: 12,
        status: 2,
        blockHeight: 245,
        fee: 0.01,
      },
    });
    expect(wrapper.vm.paymentDirection).toEqual('From: 1CK6KHY6MHgYvmRQ4PAafKYDrg1ejbH1cE');
    expect(wrapper.contains('.positive-amount')).toBe(true);
    expect(wrapper.contains('.confirmed-tx')).toBe(true);
    expect(wrapper.find('.confirmed-tx').text()).toEqual(wrapper.vm.$t('confirmed'));
    expect(wrapper.find('.tx-fee').text()).toEqual('0.01 BTC (£38.45)');
  });

  it('displays a sent Bitcoin transaction', () => {
    expect(wrapper.vm.paymentDirection).toEqual('To: 1CK6KHY6MHgYvmRQ4PAafKYDrg1ejbH1cE');
    expect(wrapper.contains('.negative-amount')).toBe(true);
    expect(wrapper.contains('.confirmed-tx')).toBe(true);
    expect(wrapper.find('.confirmed-tx').text()).toEqual(wrapper.vm.$t('confirmed'));
    expect(wrapper.find('.tx-fee').text()).toEqual('0.1 BTC (£384.52)');
  });

  it('displays a received Ethereum transaction', () => {
    router.push({ path: '/wallet/single/5' });
    wrapper.setProps({
      data: {
        amount: 3.654,
        sender: '0x8f52d186aa4a6b169abe85d999dcbe289aa215f8',
        hash: '0xdd1cb7b94395b3b0b81e38e898e2128b8e71d47c810f3450b252e216bb9039ba',
        sent: false,
        confirmations: 8,
        blockHeight: 96773,
        fee: 0.1,
      },
    });
    expect(wrapper.vm.paymentDirection).toEqual('From: 0x8f52d186aa4a6b169abe85d999dcbe289aa215f8');
    expect(wrapper.contains('.positive-amount')).toBe(true);
    expect(wrapper.contains('.unconfirmed-tx')).toBe(true);
    expect(wrapper.find('.unconfirmed-tx').text()).toEqual(wrapper.vm.$t('unconfirmed'));
    expect(wrapper.find('.tx-fee').text()).toEqual('0.1 ETH (£13.04)');
  });

  it('displays a sent Ethereum transaction', () => {
    router.push({ path: '/wallet/single/5' });
    wrapper.setProps({
      data: {
        amount: 20,
        receiver: '0x0a1443a629847f4d487cf1cf00c5b417bb27238f',
        hash: '0x629ecef1111fedc99851efdf97e94645335a8ca19361f0401601f1008cf4714c',
        sent: true,
        confirmations: -1,
        blockHeight: 2354,
        fee: 0.1,
      },
    });
    expect(wrapper.vm.paymentDirection).toEqual('To: 0x0a1443a629847f4d487cf1cf00c5b417bb27238f');
    expect(wrapper.contains('.negative-amount')).toBe(true);
    expect(wrapper.contains('.unconfirmed-tx')).toBe(true);
    expect(wrapper.find('.unconfirmed-tx').text()).toEqual(wrapper.vm.$t('pending'));
    expect(wrapper.find('.tx-fee').text()).toEqual('0.1 ETH (£13.04)');
  });

  it('displays a received ERC20 transaction', () => {
    router.push({ path: '/wallet/single/6' });
    wrapper.setProps({
      data: {
        amount: 25,
        sender: '0x8f52d186aa4a6b169abe85d999dcbe289aa215f8',
        hash: '0x9774378c83edcbf35c584f18ca8b0620b0d69cca2c8c346a963c3c7466636733',
        sent: false,
        confirmations: 130,
        blockHeight: 24539,
        fee: 0.1,
      },
    });
    expect(wrapper.vm.paymentDirection).toEqual('From: 0x8f52d186aa4a6b169abe85d999dcbe289aa215f8');
    expect(wrapper.contains('.positive-amount')).toBe(true);
    expect(wrapper.contains('.confirmed-tx')).toBe(true);
    expect(wrapper.find('.confirmed-tx').text()).toEqual(wrapper.vm.$t('confirmed'));
    expect(wrapper.find('.tx-fee').text()).toEqual('0.1 ETH (£13.04)');
  });

  it('displays a sent ERC20 transaction', () => {
    router.push({ path: '/wallet/single/6' });
    wrapper.setProps({
      data: {
        amount: 100,
        receiver: '0xcc345035d14458b3c012977f96fa1e116760d60a',
        hash: '0x4b796488cdad06be0b74bc5f3f43ba9f0def3880c6e9919a540bf8fdfb3fe015',
        sent: true,
        confirmations: -1,
        blockHeight: 87,
        fee: 0.1,
      },
    });
    expect(wrapper.vm.paymentDirection).toEqual('To: 0xcc345035d14458b3c012977f96fa1e116760d60a');
    expect(wrapper.contains('.negative-amount')).toBe(true);
    expect(wrapper.contains('.unconfirmed-tx')).toBe(true);
    expect(wrapper.find('.unconfirmed-tx').text()).toEqual(wrapper.vm.$t('pending'));
    expect(wrapper.find('.tx-fee').text()).toEqual('0.1 ETH (£13.04)');
  });
  it('displays a Litecoin transaction', () => {
    router.push({ path: '/wallet/single/7' });
    expect(wrapper.vm.paymentDirection).toEqual('To: 1CK6KHY6MHgYvmRQ4PAafKYDrg1ejbH1cE');
    expect(wrapper.contains('.negative-amount')).toBe(true);
    expect(wrapper.contains('.confirmed-tx')).toBe(true);
    expect(wrapper.find('.confirmed-tx').text()).toEqual(wrapper.vm.$t('confirmed'));
    expect(wrapper.find('.tx-fee').text()).toEqual('0.1 LTC');
  });
});
