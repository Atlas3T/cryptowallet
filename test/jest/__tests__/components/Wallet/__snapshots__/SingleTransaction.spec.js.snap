// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleTransaction.vue renders and matches snapshot 1`] = `
<div>
  <q-item-stub
    clickable="true"
    style="padding: 1rem;"
    tag="div"
  >
    <q-item-section-stub
      avatar="true"
      top="true"
    >
      <q-avatar-stub
        color="primary"
        icon="send"
        textcolor="white"
      />
    </q-item-section-stub>
     
    <q-item-section-stub>
      <q-item-label-stub
        lines="1"
      >
        
        Sent 4.97 
        
        <span>
          (£19,110.59)
        </span>
      </q-item-label-stub>
       
      <q-item-label-stub
        caption="true"
      >
        
        17 January 2019 17:27
      
      </q-item-label-stub>
    </q-item-section-stub>
     
    <q-item-section-stub
      side="true"
    >
      <q-icon-stub
        name="info"
      />
    </q-item-section-stub>
  </q-item-stub>
   
  <q-dialog-stub
    position="standard"
    transitionhide="scale"
    transitionshow="scale"
  >
    <q-card-stub>
      <q-card-section-stub
        class="transaction-details"
      >
        <div
          class="transaction-heading"
        >
          <div
            class="row justify-center"
          >
            <q-avatar-stub
              color="primary"
              icon="send"
              style="margin-bottom: 1rem;"
              textcolor="white"
            />
          </div>
           
          <div
            class="row justify-center"
          >
            
            - 4.97 
          
          </div>
           
          <div
            class="row justify-center"
          >
            
            - £19,110.59
          
          </div>
        </div>
         
        <div
          class="single-transaction-content"
        >
          <div
            class="row justify-between"
          >
            <div
              class="col-2"
            >
              
              To:
            
            </div>
             
            <div
              class="col-10"
            >
              <span
                class="tx-hash break"
              >
                
                1CK6KHY6MHgYvmRQ4PAafKYDrg1ejbH1cE
              
              </span>
               
              <q-btn-stub
                align="center"
                dense="true"
                flat="true"
                icon="content_copy"
                ripple="true"
                round="true"
                size="xs"
              />
            </div>
          </div>
           
          <div
            class="row justify-between"
          >
            <div
              class="col-2"
            >
              
              Hash:
            
            </div>
             
            <div
              class="col-10"
            >
              <span
                class="tx-hash break"
              >
                
                d02995c1faf572c5d9ae966915aa7958c88ed2e47f8f208b68d82d863bf61010
                
                <q-btn-stub
                  align="center"
                  dense="true"
                  flat="true"
                  icon="content_copy"
                  ripple="true"
                  round="true"
                  size="xs"
                />
              </span>
            </div>
          </div>
           
          <div
            class="row justify-between"
          >
            <div>
              
              Status:
            
            </div>
             
            <div>
              <span
                class="status confirmed-tx"
              >
                
                Confirmed
              
              </span>
            </div>
          </div>
           
          <div
            class="row justify-between"
          >
            <div>
              
              Confirmations:
            
            </div>
             
            <div>
              <span
                class="confirmations"
              >
                
                12
              
              </span>
            </div>
          </div>
           
          <div
            class="row justify-between"
          >
            <div>
              
              Fee:
            
            </div>
             
            <div>
              <span
                class="tx-fee"
              >
                
                0.1 BTC (£384.52)
              
              </span>
            </div>
          </div>
           
          <div
            class="row justify-between"
          >
            <div>
              
              Block Height:
            
            </div>
             
            <div>
              <span
                class="tx-height"
              >
                
                245
              
              </span>
            </div>
          </div>
           
          <div
            class="row justify-end"
          >
            <q-btn-stub
              align="center"
              dense="true"
              flat="true"
              label="Close"
              ripple="true"
              round="true"
            />
          </div>
        </div>
      </q-card-section-stub>
    </q-card-stub>
  </q-dialog-stub>
</div>
`;
